spark3-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestDelete.java

```diff
@@ -721,9 +723,25 @@ public void testDeleteRefreshesRelationCache() throws NoSuchTableException {
 
   // TODO: multiple stripes for ORC
 
+  protected void setDistributionModeOnTable() {
+    DistributionMode distMode = DistributionMode.NONE;
+    switch (fileFormat) {
+      case "parquet":
+        distMode = DistributionMode.HASH;
+        break;
+      case "orc":
+        break;
+      case "avro":
+        distMode = DistributionMode.RANGE;
+        break;
+    }
+    sql("ALTER TABLE %s SET TBLPROPERTIES('%s' '%s')", tableName, WRITE_DISTRIBUTION_MODE, distMode);
```

Comment:
This produces the same matrix of parameters as in #2911, but only for the partitioned tables used in TestDelete and its subclass(es).

Reply:
@wypoon Maybe you can just move these modify in #2911  to this patch. It can also test `UPDATE` and `MERGE into` in different distribution modes.

Reply:
@wypoon Maybe you can just move these modify in #2911  to this patch. It can also test `UPDATE` and `MERGE into` in different distribution modes.

Reply:
@coolderli #2954 (which doesn't have test changes) got merged instead. I look forward to your completing #2911.